packages/wallet-sdk/src/core/communicator/Communicator.ts

```diff
@@ -16,17 +17,19 @@ import { closePopup, openPopup } from ':util/web';
 export class Communicator {
   static communicators = new Map<string, Communicator>();
 
+  private readonly metadata: AppMetadata;
   private readonly url: URL;
   private popup: Window | null = null;
   private listeners = new Map<(_: MessageEvent) => void, { reject: (_: Error) => void }>();
 
-  private constructor(url: string = CB_KEYS_URL) {
+  private constructor(url: string = CB_KEYS_URL, metadata: AppMetadata) {
     this.url = new URL(url);
+    this.metadata = metadata;
```

Comment:
i wonder whether this needs to be a param of `waitForPopupLoaded` method, not in constructor level.
(btw, we might rename the method to be, initialize or connect maybe??)
@cb-jake

Reply:
@bangtoven If we made it a param of `waitForPopupLoaded` we would probably need to make it a param `postMessage` as well which would require passing it on calls to `postMessage`. Currently we call `waitForPopupLoaded` in 2 places, though one of those places is inside the communicators postmessage method which is called in 4 spots. I'm not positive we would have access to the metadata in all spots we call these methods